File: laythe_vm/fixture/rb_benchmark/equality.rb
# frozen_string_literal: true

loopStart = Time.new

10000000.times do |_|
  1; 1; 1; 2; 1; nil; 1; "str"; 1; true;
  nil; nil; nil; 1; nil; "str"; nil; true;
  true; true; true; 1; true; false; true; "str"; true; nil;
  "str"; "str"; "str"; "stru"; "str"; 1; "str"; nil; "str"; true;
end

loopTime = Time.new - loopStart

start = Time.new

10000000.times do |_|
  1 == 1; 1 == 2; 1 == nil; 1 == "str"; 1 == true;
  nil == nil; nil == 1; nil == "str"; nil == true;
  true == true; true == 1; true == false; true == "str"; true == nil;
  "str" == "str"; "str" == "stru"; "str" == 1; "str" == nil; "str" == true;
end

elapsed = Time.new - start;
puts "loop"
puts loopTime
puts "elapsed"
puts elapsed
puts "equals"
puts elapsed - loopTime

File: laythe_vm/fixture/py_benchmark/equality.py
import time

i = 0

loopStart = time.time()

for _ in range(10000000):
  1; 1; 1; 2; 1; None; 1; "str"; 1; True
  None; None; None; 1; None; "str"; None; True
  True; True; True; 1; True; False; True; "str"; True; None
  "str"; "str"; "str"; "stru"; "str"; 1; "str"; None; "str"; True


loopTime = time.time() - loopStart

start = time.time()

for _ in range(10000000):
  1 == 1; 1 == 2; 1 == None; 1 == "str"; 1 == True
  None == None; None == 1; None == "str"; None == True
  True == True; True == 1; True == False; True == "str"; True == None
  "str" == "str"; "str" == "stru"; "str" == 1; "str" == None; "str" == True

elapsed = time.time() - start
print("loop")
print(loopTime)
print("elapsed")
print(elapsed)
print("equals")
print(elapsed - loopTime)
